### PR TITLE
implement calculator with TestCalculator

### PR DESCRIPTION
**Pull request by IwalkInSilence**

### CODE REVIEW

File: src/main/java/core/basesyntax/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        if (operation.equals("*")) {
+            return firstValue * secondValue;
+        }
+        if (operation.equals("/")) {
+            if (secondValue == 0) {
+                throw new ArithmeticException("You can't division on zero!");
+            }
+            return firstValue / secondValue;
+        }
+        if (operation.equals("^")) {
```
Comment (Artificial-Friend) [inline src/main/java/core/basesyntax/Calculator.java:20]: If I need to get here I will redundantly invoke 4 other if() statements. Better change it to another condition statement.

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class CalculatorTest {
+    private static final double DELTA = 0.000001;
+    Calculator calculator = new Calculator();
```
Comment (Artificial-Friend) [inline src/test/java/core/basesyntax/CalculatorTest.java:10]: It's better to initialize it in method that will run before all tests.

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        assertEquals(-140.5, calculator.calculate(-25.5, -115, "+"), DELTA);
+        assertEquals(4.37, calculator.calculate(5.35, -0.98, "+"), DELTA);
+        assertEquals(-129, calculator.calculate(0, -129, "+"), DELTA);
+        assertEquals(10025, calculator.calculate(10025, 0, "+"), DELTA);
+        assertEquals(Double.POSITIVE_INFINITY, calculator.calculate(
+                Double.MAX_VALUE, Double.MAX_VALUE, "+"), DELTA);
+    }
+
+    @Test
+    void subtractionTests_Ok() {
```
Comment (Artificial-Friend) [inline src/test/java/core/basesyntax/CalculatorTest.java:24]: Would be nice to have operations with MIN and MAX values here. Since you're using double INFINITY is a legit value for method and it can accept it. I suppose you need to test it too.

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        assertEquals(89.5, calculator.calculate(-25.5, -115, "-"), DELTA);
+        assertEquals(6.33, calculator.calculate(5.35, -0.98, "-"), DELTA);
+        assertEquals(129, calculator.calculate(0, -129, "-"), DELTA);
+        assertEquals(10025, calculator.calculate(10025, 0, "-"), DELTA);
+        assertEquals(0, calculator.calculate(Double.MIN_VALUE, Double.MIN_VALUE, "-"), DELTA);
+
+    }
+
+    @Test
+    void multiplicationTests_Ok() {
```
Comment (Artificial-Friend) [inline src/test/java/core/basesyntax/CalculatorTest.java:35]: Same here with MIN, MAX and INFINITY.
Also INFINITY could have two states, be sure to check if your method work correct with it.

**Review by Artificial-Friend — CHANGES_REQUESTED** [review]
Since you're using double it's possible to get NaN, I recommend to add this kind of tests and exception.

File: src/main/java/core/basesyntax/Calculator.java (hunk 1 above)
Comment (IwalkInSilence) [reply src/main/java/core/basesyntax/Calculator.java]: change all calculator on switch

File: src/test/java/core/basesyntax/CalculatorTest.java (hunk 2 above)
Comment (IwalkInSilence) [reply src/test/java/core/basesyntax/CalculatorTest.java]: change but it's not necessary as Bodya said

File: src/test/java/core/basesyntax/CalculatorTest.java (hunk 3 above)
Comment (IwalkInSilence) [reply src/test/java/core/basesyntax/CalculatorTest.java:32]: added methods

File: src/test/java/core/basesyntax/CalculatorTest.java (hunk 4 above)
Comment (IwalkInSilence) [reply src/test/java/core/basesyntax/CalculatorTest.java:47]: added tests

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 5]
```diff
@@ -28,7 +36,11 @@ void subtractionTests_Ok() {
         assertEquals(129, calculator.calculate(0, -129, "-"), DELTA);
         assertEquals(10025, calculator.calculate(10025, 0, "-"), DELTA);
         assertEquals(0, calculator.calculate(Double.MIN_VALUE, Double.MIN_VALUE, "-"), DELTA);
-
+        assertEquals(0, calculator.calculate(Double.MAX_VALUE, Double.MAX_VALUE, "-"), DELTA);
+        assertEquals(Double.NaN, calculator.calculate(
```
Comment (Artificial-Friend) [inline src/test/java/core/basesyntax/CalculatorTest.java:40]: I would move this to another custom test rather than having it in tests that comparing legal values.

**Review by Artificial-Friend — APPROVED** [review]

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class CalculatorTest {
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:9]: add access modifier to the class and to all the methods

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        assertEquals(0.04, calculator.calculate(5, -2, "^"), DELTA);
+        assertEquals(-100000, calculator.calculate(-10, 5, "^"), DELTA);
+        assertEquals(0.87055, calculator.calculate(2, -0.2, "^"), DELTA);
+        assertEquals(0.00390625, calculator.calculate(0.5, 8, "^"), DELTA);
+        assertEquals(Double.POSITIVE_INFINITY, calculator.calculate(
+                Double.MAX_VALUE, Double.MAX_VALUE, "^"), DELTA);
+        assertEquals(Double.POSITIVE_INFINITY, calculator.calculate(
+                Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY, "^"), DELTA);
+        assertEquals(0, calculator.calculate(
+                Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY, "^"), DELTA);
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:95]: let's add tests for raising to zero power

and raising 0 to some power

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:96]: redundant empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]
nice job)
